instructors.Rmd

```diff
@@ -0,0 +1,294 @@
+---
+layout: page
+title: Programming with R
+subtitle: Instructor's Guide
+minutes: 0
+---
+
+```{r, include = FALSE}
+source("tools/chunk-options.R")
+```
+
+## Legend
+
+FIXME
+
+## Overall
+
+This lesson is written as an introduction to R, but its real purpose is to
+introduce the single most important idea in programming: how to solve problems
+by building functions, each of which can fit in a programmer's working memory.
+In order to teach that, we must teach people a little about the mechanics of
+manipulating data with lists and file I/O so that their functions can do things
+they actually care about.
+Our teaching order tries to show practical uses of every idea as soon as it is
+introduced; instructors should resist the temptation to explain the "other 90%"
+of the language as well.
+
+The secondary goal of this lesson is to give them a usable mental model of how
+programs run (what computer science educators call a
+[notional machine](reference.html#notional-machine) so that they can debug
+things when they go wrong.
+In particular, they must understand how function call stacks work.
+
+The final example asks them to build a command-line tool that works with the
+Unix pipe-and-filter model.
+We do this because it is a useful skill and because it helps learners see that
+the software they use isn't magical.
+Tools like `grep` might be more sophisticated than the programs our learners can
+write at this point in their careers, but it's crucial they realize this is a
+difference of scale rather than kind.
+
+A typical, half-day, lesson will use the following material:
```

Comment:
This is too much material for a half day. We should not give them unrealistic expectations. In a half-day, it is reasonable to cover the first three lessons if they do not get too bogged down in the functions lesson (this will depend on the previous experience of the learners). A full day should be budgeted if the goal is to cover the entire lesson.